core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -182,6 +182,7 @@
     public static final String FORCE_SPILLING_JOIN = "force_spilling_join";
     public static final String FAULT_TOLERANT_EXECUTION_FORCE_PREFERRED_WRITE_PARTITIONING_ENABLED = "fault_tolerant_execution_force_preferred_write_partitioning_enabled";
     public static final String PAGE_PARTITIONING_BUFFER_POOL_SIZE = "page_partitioning_buffer_pool_size";
+    public static final String IGNORE_BROKEN_CATALOGS = "ignore_broken_catalogs";
```

Comment:
It can be server config instead of per-session. Per-session is a bad behaviour when it changes the results in such big way.

Reply:
I think you probably don't want to ignore such errors in a data pipeline while being more lenient in bi tools, so I think being able to drive this from the client is more logic.

Reply:
I do not feel convinced. I think we should single vision where and how manifest connector errors. For example we do suppress errors when querying `information_schema` and listing object in JDBC is to me very the same.

Also to go further what if the error is transient? Then ignoring that error may cause poor UX that sometimes tables/schemas are visible and sometimes are not.

Reply:
Just logging something would be the same as ignoring the error from a BI tools perspective.

I guess we need alignment before the work on this PR can continue.

Reply:
@martint @electrum  @findepi @hashhar  What do you think about the above? Should JDBC object listing methods behave like query against `information_schema`? I mean,  should they not throw an error if the connector is misconfigured, but return empty list instead?

Notice that JDBC Metadata methods translate to SQL like:
```
        StringBuilder query = new StringBuilder("" +
                "SELECT TABLE_CAT, TABLE_SCHEM, TABLE_NAME, TABLE_TYPE, REMARKS,\n" +
                "  TYPE_CAT, TYPE_SCHEM, TYPE_NAME, " +
                "  SELF_REFERENCING_COL_NAME, REF_GENERATION\n" +
                "FROM system.jdbc.tables");
```

So in my opinion it should be coherent with they way `information_schema.tables` is implemented. And it looks like these tables are both using `MetadataManager` so the problem you are solving seems to be affecting `information_schema` too.

Reply:
> I think you probably don't want to ignore such errors in a data pipeline while being more lenient in bi tools, so I think being able to drive this from the client is more logic.

makes sense to me


> I do not feel convinced. I think we should single vision where and how manifest connector errors. For example we do suppress errors when querying `information_schema` and listing object in JDBC is to me very the same.

agreed

> Also to go further what if the error is transient? Then ignoring that error may cause poor UX that sometimes tables/schemas are visible and sometimes are not.

yes, that's unavoidable when you continue execution in a presence of an error

> Should JDBC object listing methods behave like query against information_schema? 

i think so

> should they not throw an error if the connector is misconfigured, but return empty list instead?

i am not aware under what circumstances information_schema behaves like that
i think @losipiuk attempted to capture the desired long-term state https://github.com/trinodb/trino/issues/6551#issuecomment-903798696
but it would be a big project to get us there.

Reply:
The name `ignore_broken_catalogs` needs to be improved.  It is not clear from the name what this effects and what it does not.  The name to let the user know the scope of "ignore".